### PR TITLE
Better structure VMR join points for resolvers

### PR DESCRIPTION
**Pull request by ViktorHofer**
Follow-up on https://github.com/dotnet/source-build/issues/4691

This change better encapsulates the builds of various components so that those don't rely on the sdk layout generation targets anymore.

- `VSMSBuildExtensions`

Get the assemblies for Microsoft.Deployment.DotNet.Releases and hostfxr (from the runtime packs) from the `Microsoft.DotNet.MSBuildSdkResolver` dependency instead of downloading and copying them again. This makes sure that these dependencies are in sync between VSMSBuildExtensions and its own dependencies.

Also simplify packaging by removing unnecessary conditions and change the TFM to net472 as this is a VS only package.

- `VSTemplateLocator`

Let VSTemplateLocator use the same packaging infrastructure as VSMSBuildExtensions (use dotnet pack / Pack target instead of invoking NuGet.exe via a Powershell script manually). Delete the now unnecessary nuspec file.

Same as for VSMSBuildExtensions, consujme the hostfxr dependencies from the `Microsoft.DotNet.TemplateLocator` dependency to avoid a disjoin. The `Microsoft.Deployment.DotNet.Releases` dependeny is still kept as that one isn't referenced by project dependencies.

- Remove an outdated comment about `Microsoft.DotNet.MSBuildSdkResolver` in SetBuildDefaults.targets.

- Remove `Microsoft.DotNet.TemplateLocator`'s package which isn't used by dotnet/templating (and any other non-shipping component in our stack) anymore. Simplify the infrastructure after removing the package and correctly condition the project to only build the .NET Framework TFM inside the VMR in the second build pass.

- `Microsoft.DotNet.MSBuildSdkResolver`

Remove the .NETCoreApp TFM from `Microsoft.DotNet.MSBuildSdkResolver` which isn't used anymore. All the responsible logic for resolving inbox SDKs got moved into [Microsoft.DotNet.SdkResolver](https://github.com/dotnet/sdk/tree/main/src/Resolvers/Microsoft.DotNet.SdkResolver) years ago.

Don't build that project when building from source as it's .NET Framework only.

Stop creating a non-shipping package for it as it isn't used anymore. VS gets it via the VSMSBuildExtensions package.

- `WorkloadManifestReader`

Always build all the TFMs regardless of the build OS as other libraries touched by this change depend on that library. Apply small clean-up.

- Update the MSBuildSdkResolver tests to only target .NETFramework and update UnitTests.proj to allow projects to only target .NET Framework.

---

Here's a small description of how resolvers get built for the two VS transport packages:

```
VSMSBuildExtensions -> Microsoft.NET.Build.Extensions.Tasks (net472)
                    -> Microsoft.DotNet.MSBuildSdkResolver!

VSTemplateLocator   -> Microsoft.DotNet.TemplateLocator (net472) -> Microsoft.DotNet.SdkResolver! (net472)
```

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Diagnosing those .NET Framework test failures is tricky. What I understand is:

The test failures happen when running with desktop msbuild. The desktop msbuild is very old, from May 29th and only has binding redirects for STJ up to 8.0.0.3.

What I absolutely don't understand is how my change could cause a new dependency on STJ/8.0.0.4 while this works just fine in main.

File: src/Microsoft.DotNet.TemplateLocator/Microsoft.DotNet.TemplateLocator.csproj [hunk 1]
```diff
@@ -2,30 +2,17 @@
...
     <StrongNameKeyId>MicrosoftAspNetCore</StrongNameKeyId>
-    <IsPackable>true</IsPackable>
     <!-- Create FileDefinitions items for ResolveHostfxrCopyLocalContent target -->
     <EmitLegacyAssetsFileItems>true</EmitLegacyAssetsFileItems>
-    <PackageId>microsoft.dotnet.templateLocator</PackageId>
-    <Nullable>Enable</Nullable>
-
+    <Nullable>enable</Nullable>
     <!-- https://github.com/dotnet/sdk/issues/14801 -->
-    <AssemblyVersion Condition="'$(TargetFramework)'=='net472'">5.0.100.0</AssemblyVersion>
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.TemplateLocator/Microsoft.DotNet.TemplateLocator.csproj:17]: the issue was closed, I guess we should follow up whether this is still needed.

File: src/Installer/redist-installer/projects/VSMSBuildExtensions/VSMSBuildExtensions.csproj [hunk 2]
```diff
@@ -65,10 +51,10 @@
...
+      <VSMSBuildExtensionsContent Include="$(ArtifactsBinDir)Microsoft.DotNet.MSBuildSdkResolver\$(Configuration)\net472\x86\hostfxr.dll" DeploymentSubpath="MSBuildSdkResolver/x86/" />
+      <VSMSBuildExtensionsContent Include="$(ArtifactsBinDir)Microsoft.DotNet.MSBuildSdkResolver\$(Configuration)\net472\x64\hostfxr.dll" DeploymentSubpath="MSBuildSdkResolver/x64/" />
+      <VSMSBuildExtensionsContent Include="$(ArtifactsBinDir)Microsoft.DotNet.MSBuildSdkResolver\$(Configuration)\net472\arm64\hostfxr.dll" DeploymentSubpath="MSBuildSdkResolver/arm64/" />
+      <VSMSBuildExtensionsContent Include="$(ArtifactsBinDir)Microsoft.DotNet.MSBuildSdkResolver\$(Configuration)\net472\**\Microsoft.Deployment.DotNet.Releases*.dll" DeploymentSubpath="MSBuildSdkResolver/" />
       <VSMSBuildExtensionsContent Include="$(ArtifactsBinDir)Microsoft.DotNet.MSBuildSdkResolver\$(Configuration)\net472\**\Microsoft.DotNet.MSBuildSdkResolver*.dll" DeploymentSubpath="MSBuildSdkResolver/" />
```
Comment (akoeplinger) [inline src/Installer/redist-installer/projects/VSMSBuildExtensions/VSMSBuildExtensions.csproj:58]: do you know why we use `*` in the filename here? from what I can see these only match one file respectively.

**Review by akoeplinger — APPROVED** [review]

File: src/Installer/redist-installer/projects/VSMSBuildExtensions/VSMSBuildExtensions.csproj (hunk 2 above)
Comment (ViktorHofer) [reply src/Installer/redist-installer/projects/VSMSBuildExtensions/VSMSBuildExtensions.csproj:58]: Yes, that's to include satellite assemblies for localization.

**Pull request merged**